File: rb/features.rb
# WeatherData SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module WeatherDataFeatures
  def self.make_feature(name)
    case name
    when "base"
      WeatherDataBaseFeature.new
    when "ratelimit"
      WeatherDataRatelimitFeature.new
    when "retry"
      WeatherDataRetryFeature.new
    when "test"
      WeatherDataTestFeature.new
    when "timeout"
      WeatherDataTimeoutFeature.new
    else
      WeatherDataBaseFeature.new
    end
  end
end

File: py/features.py
# WeatherData SDK feature factory

from feature.base_feature import WeatherDataBaseFeature
from feature.test_feature import WeatherDataTestFeature


def _make_feature(name):
    features = {
        "base": lambda: WeatherDataBaseFeature(),
        "test": lambda: WeatherDataTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
